includes/blocks/site-automation-block/edit.js

```diff
@@ -25,26 +36,234 @@ import { editPropsShape } from './props-shape';
  * @param {boolean}  props.attributes.displayFeaturedImage   Whether to display featured image.
  * @param {boolean}  props.attributes.addLinkToFeaturedImage Whether to add post permalink to featured image.
  * @param {string}   props.className                         Class name for the block.
+ * @param {string}   props.clientId                          Block's client ID.
  * @param {Function} props.setAttributes                     Sets the value for block attributes.
  *
- * @return {Function} Render the edit screen
+ * @returns {Function} Render the edit screen
  */
 const SiteAutomationBlockEdit = ({
 	attributes: {
-		pageName,
-		widgetName,
-		displayPostExcept,
-		displayAuthor,
-		displayPostDate,
-		displayFeaturedImage,
-		addLinkToFeaturedImage,
+		pageName = '',
+		widgetName = '',
+		displayPostExcept = false,
+		displayAuthor = false,
+		displayPostDate = false,
+		displayFeaturedImage = false,
+		addLinkToFeaturedImage = false,
 	},
 	className,
 	setAttributes,
+	clientId,
 }) => {
+	const blockProps = useBlockProps();
+	const [message, setMessage] = useState({
+		text: '',
+		color: 'green',
+	});
+	const { replaceInnerBlocks } = dispatch(blockEditorStore);
+	const ALLOWED_BLOCKS = ['sophi/page-list-item'];
+
+	// Subscribe to the innerBlocks' state changing.
+	const innerBlocks = useSelect((select) => {
+		return select(blockEditorStore).getBlock(clientId).innerBlocks;
+	});
+
+	const sophiEndpoint = '/sophi/v1/';
+
+	const curatorPosts = async (overridePostID = '') => {
+		if (pageName === '' || widgetName === '') {
+			return;
+		}
+
+		const queryArgs = {
+			overridePostID,
+			pageName,
+			widgetName,
+			displayFeaturedImage,
+			displayAuthor,
+			displayPostDate,
+			displayPostExcept,
+		};
+
+		const updatedInnerBlocks = [];
+
+		await apiFetch({
+			path: addQueryArgs(`${sophiEndpoint}site-automation`, {
+				...queryArgs,
+			}),
+			method: 'GET',
+		}).then(
+			// eslint-disable-next-line consistent-return
+			(data) => {
+				if (!data?.length) {
+					setMessage({
+						text: __('Posts not found in the site!', 'sophi-wp'),
+						color: 'red',
+					});
+					return false;
+				}
+
+				// eslint-disable-next-line array-callback-return
+				data.map((item) => {
```

Comment:
if you don't care about the returned value you can also use `.forEach` here. It doesn't care about a return value